rust_src/src/buffers.rs

```diff
@@ -434,4 +474,36 @@ pub extern "C" fn nsberror(spec: Lisp_Object) -> ! {
     }
 }
 
+/// These functions are for debugging overlays.
+
+/// Return a pair of lists giving all the overlays of the current buffer.
+/// The car has all the overlays before the overlay center;
+/// the cdr has all the overlays after the overlay center.
+/// Recentering overlays moves overlays between these lists.
+/// The lists you get are copies, so that changing them has no effect.
+/// However, the overlays you get are the real objects that the buffer uses.
+#[lisp_fn]
+pub fn overlay_lists() -> LispObject {
+    let list_overlays = |ol: Option<LispOverlayRef>| -> LispObject {
+        let mut ol_list = LispObject::constant_nil();
+
+        if let Some(o) = ol {
+            let mut it = o.iter();
+
+            while let Some(n) = it.next() {
+                let tmp: LispObject =
+                    unsafe { mem::transmute(LispObject::tag_ptr(n, Lisp_Type::Lisp_Misc)) };
+                ol_list = unsafe { LispObject::from_raw(Fcons(tmp.to_raw(), ol_list.to_raw())) };
```

Comment:
Since everything is a `LispObject` wouldn't `LispObject::cons` work better here?

Reply:
I have to cast it to `Lisp_Object` for the next call of `Fcons`. Should I add a method `to_raw` for `Lisp_Cons` ?